pkg/services/control/server/list_objects.go

```diff
@@ -0,0 +1,59 @@
+package control
+
+import (
+	"errors"
+
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine"
+	"github.com/nspcc-dev/neofs-node/pkg/services/control"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (s *Server) ListObjects(req *control.ListObjectsRequest, stream control.ControlService_ListObjectsServer) error {
+	// verify request
+	if err := s.isValidRequest(req); err != nil {
+		return status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	// check availability
+	err := s.ready()
+	if err != nil {
+		return status.Error(codes.Internal, err.Error())
+	}
+
+	var prm engine.ListWithCursorPrm
+	prm.WithCount(1024)
+	for {
+		res, err := s.storage.ListWithCursor(prm)
+		if err != nil {
+			if errors.Is(err, engine.ErrEndOfListing) {
+				return nil
+			}
+
+			return status.Error(codes.Internal, err.Error())
+		}
+
+		addresses := res.AddressList()
+		objectsAddresses := make([][]byte, 0, len(addresses))
+		for _, objectId := range addresses {
+			objectsAddresses = append(objectsAddresses, []byte(objectId.Address.EncodeToString()))
+		}
+
+		resp := &control.ListObjectsResponse{
+			Body: &control.ListObjectsResponse_Body{
+				ObjectAddress: objectsAddresses,
```

Comment:
now u propose to transmit 1024 of 89B addresses per message, which is ~89K. It's still far away from available 4MB limit, so we gonna sign and send more messages then we could

lets reserve 64K for service bytes and future fields, and fit soft limit `=(4M-64K)/89=46390` of addresses. Ofc write a comment how this number is chosen

the next optimization could be packing addresses into the message w/o intermediate `addresses := res.AddressList()` memory buffer. `engine.ListWithCursorPrm` could accept functional address handler to make it possible. I suggest to leave an issue for this

Reply:
Done.